include/eve/arch/cpu/tags.hpp

```diff
@@ -33,26 +34,39 @@ namespace eve
   struct aggregated_ : cpu_
   {
     using parent = cpu_;
+    static constexpr bool regular_logical_register = true;
   };
 
   //================================================================================================
-  // Concept for discriminatig aggregated from non-agrgegated ABI
+  // Concept for discriminatig aggregated from non-agregated ABI
+  //================================================================================================
   template<typename T> concept regular_abi = !std::same_as<T,aggregated_>;
 
   //================================================================================================
   // Dispatching tag for emulated SIMD implementation
   struct emulated_ : cpu_
   {
-    static constexpr std::size_t bits           = 128;
-    static constexpr std::size_t bytes          = 16;
-    static constexpr bool        is_bit_logical = true;
+    static constexpr std::size_t bits                     = 128;
+    static constexpr std::size_t bytes                    = 16;
+    static constexpr bool        regular_logical_register = true;
 
     template<typename Type>
     static constexpr std::size_t expected_cardinal = bytes / sizeof(Type);
   };
 
+  template<typename T>
+  concept non_native_abi = detail::is_one_of<T>(detail::types<aggregated_, emulated_> {});
+
+  template<typename T>
+  concept native_abi = !detail::is_one_of<T>(detail::types<aggregated_, emulated_> {});
+
+  //================================================================================================
+  // Checks for logical status in ABI
+  template<typename ABI>
+  struct use_regular_logical_register : std::integral_constant<bool,ABI::regular_logical_register>
+  {};
```

Comment:
Is this better for error messages then a typedef?

Reply:
That's classic meta-function forwarding style.